lib/OpenQA/Schema/Result/Assets.pm

```diff
@@ -116,11 +116,19 @@ sub remove_from_disk {
 
     my $type = $self->type;
     my $name = $self->name;
-    my $file = locate_asset($type, $name, mustexist => 1);
+    my $file = $self->{_location} = locate_asset($type, $name, mustexist => 1);
     if (!defined $file) {
         log_info("GRU: skipping removal of $type/$name; it does not exist anyways");
         return undef;
     }
+    # prevent removal of fixed assets
+    # note: The asset might have been moved into the "fixed" directory after it
+    #       was initially tracked by openQA. We do not re-check whether an asset is
+    #       fixed on every cleanup.
+    if ($self->is_fixed) {
+        log_info("GRU: skipping removal of $type/$name; it is fixed");
```

Comment:
Does this info message appear more than once or every time we try to delete the asset in vain? If it reappears I would use debug, not info, but then the question would be if we need to prevent the call of remove at all

Reply:
It will not re-appear because we will still delete the asset in the database. Then, on the next re-scan it'll be tracked as fixed asset and we won't reach this code. But I admit that this is a bit messy.